File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.lang.*;
import java.util.*;
import java.io.*;
class Solution {
    public static boolean[] visited;
    
    public static void dfs(int node, int[][] computers){
        visited[node]=true;
        for(int i=0;i<computers[node].length;i++){
            if(computers[node][i]==0||visited[i]) continue;
            dfs(i, computers);
        }
    }
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited=new boolean[n];
        Arrays.fill(visited, false);
        for(int i=0;i<n;i++){
            if(visited[i]) continue;
            dfs(i, computers);
            answer++;
        }    
        return answer;
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;
void dfs(int node, vector<bool>& visited, vector<vector<int>> computers, int& cnt){
    cnt++;
    visited[node]=true;
    for(int i=0;i<computers[node-1].size();i++){  //i+1번째 노드로 갈 수 있는가
        if(computers[node-1][i]==1){
            if(!visited[i+1]) dfs(i+1, visited, computers, cnt); 
        }
    }
    return;
}
int solution(int n, vector<vector<int>> computers) {
    vector<bool> visited(n+1, false);
    int answer = 0;
    for(int i=1;i<=n;i++){ //node순회
        if(visited[i]) continue;
        int cnt=0;
        dfs(i, visited, computers, cnt);
        //if(cnt>1) answer++;
        answer++;
    }
    return answer;
}